### ISSUE TITLE
To-do list for the remainder

### ISSUE THREAD

**Issue by jpivarski** [hsf-training/deep-learning-intro-for-hep#7]
Based on [CoDaS-HEP lesson 2](https://github.com/jpivarski-talks/2024-07-24-codas-hep-ml/blob/main/lesson-2-issues/lecture-slides.ipynb):

- [x] Minimizing the objective function
- [x] Epochs and mini-batches
  - [x] Maybe another exercise: optimizing the mini-batch size
- [x] Feature selection and the "kernel trick"
- [x] Under & overfitting
- [x] Regularization: L1, L2, dropout
  - [x] Exercise: L1 & L2 regularization in TensorFlow Playground
- [x] Hyperparameters, validation, and partitioning into training, validation, and test samples
- [x] Goodness of fit metrics

And then

- [x] [The project](https://github.com/jpivarski-talks/2024-07-24-codas-hep-ml/blob/main/lesson-3-project/description-and-setup.ipynb)

And finally, from [CoDaS-HEP lesson 4](https://github.com/jpivarski-talks/2024-07-24-codas-hep-ml/blob/main/lesson-4-architectures/lecture-slides.ipynb):

- [x] Unsupervised problems and problems beyond regression & classification
- [x] Autoencoders (example of unsupervised)
- [x] Convolutional neural networks (with the jet images)
- [x] DeepSets (example of ragged data)

It's not clear how much I should get into these topics. Perhaps I should stop after these three and leave it open for others to add more.

**Comment by jpivarski** [comment]
Okay, now it's done.
